### PR TITLE
Handle 404 page

### PR DESCRIPTION
**Pull request by ciaranschutte**
- adds our 404 page at app level to handle 404s including those caused by `notFound()`

### CODE REVIEW

File: src/app/(post-login)/submission/program/page.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ import PROGRAMS_LIST_QUERY from '@/app/gql/PROGRAMS_LIST_QUERY';
 import { notNull } from '@/global/utils/types';
 import { useQuery } from '@apollo/client';
 import { notFound } from 'next/navigation';
-import ProgramList from './components/ProgramList';
+import ProgramList from '../components/ProgramList';
```
Comment (justincorrigible) [inline src/app/(post-login)/submission/program/page.tsx:27]: For consistency...
```suggestion
import ProgramList from '@/components/ProgramList';
```

**Review by justincorrigible — APPROVED** [review]
Couple comments about absolute vs relative paths, for consistency... but this can be squashed + merged as is

File: src/app/(post-login)/submission/program/page.tsx (hunk 1 above)
Comment (ciaranschutte) [reply src/app/(post-login)/submission/program/page.tsx:27]: This is correct.it is confusing organisation pattern.
Leaning heavily into Next colocation a lot of pages have a `components` folder for specific uses of a component but there's also a components folder in the root at `@/components` for general use

File: src/app/not-found.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ */
+
+'use client';
+
+import { useTheme } from '@/lib/emotion';
+import { css, Link, Typography } from '@icgc-argo/uikit';
+import Image from 'next/image';
+import NextLink from 'next/link';
+import { Col, Row } from 'react-grid-system';
+import dnaBrokenImage from '../../public/assets/dna-broken.svg';
```
[the remark this answers is not in the document]
Comment (ciaranschutte) [reply src/app/not-found.tsx:27]: good idea. ty

File: src/app/(post-login)/submission/program/page.tsx (hunk 1 above)
Comment (justincorrigible) [reply src/app/(post-login)/submission/program/page.tsx:27]: wait, so you mean there's a `src/app/(post-login)/submission/components`? 👎

Comment (ciaranschutte) [reply src/app/(post-login)/submission/program/page.tsx:27]: that's exactly what I mean.
any suggestions on improving that organisation?

Comment (ciaranschutte) [reply src/app/(post-login)/submission/program/page.tsx:27]: one option is to remove the "component" folder but then you have a bunch of uncatagorised files in a single folder

**Pull request merged**